[Event listener] Add a new supervisord event listener to pmon container

keboliu opened the pull request:
Event-listener: Add a new supervisord event listener to pmon container to clear the state DB  when pmon was shut down.

* This event listener will monitor the 'SUPERVISOR_STATE_CHANGE_STOPPING' event of supervisord when it is shutting down.   On this event, the event listener will call command 'post-syseeprom -c' to clear the state DB.

Signed-off-by: Kebo Liu kebol@mellanox.com

jleveque commented:
Since this is specific to the PMON container, why not add the event listener script in https://github.com/Azure/sonic-buildimage/tree/master/dockers/docker-platform-monitor?

keboliu commented:
> Since this is specific to the PMON container, why not add the event listener script in https://github.com/Azure/sonic-buildimage/tree/master/dockers/docker-platform-monitor?

I was thinking to do so but not quite sure whether this is a good practice at that time. Since you all prefer this way, I moved it and updated PR https://github.com/Azure/sonic-buildimage/pull/2866
